docker/docker.go

```diff
@@ -108,8 +108,12 @@ func NewImage(qname, user, password string, insecureTLS bool) (*Image, error) {
 	if name == "" {
 		name = strings.Join(nameParts, "/")
 	}
+	if insecureRegistry {
+		registry = fmt.Sprintf("http://%s/v2", registry)
+	} else {
+		registry = fmt.Sprintf("https://%s/v2", registry)
```

Comment:
By insecure, do you mean self-signed/invalid certificates?
This patch allows for HTTP-only registry, without SSL at all.